src/main/java/hudson/tasks/Ant.java

```diff
@@ -190,14 +190,21 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListene
 
         args.addKeyValuePairs("-D",build.getBuildVariables(),sensitiveVars);
 
-        args.addKeyValuePairsFromPropertyString("-D",properties,vr,sensitiveVars);
+        //[cs] to support expandableTextboxes, single-liner properties need to be converted into
+        // newline separated properties.
+        String properties2 = properties;
+        if (!properties.contains("\n")) {
+        	properties2 = properties.replaceAll(" +", "\n");
+        }
+        args.addKeyValuePairsFromPropertyString("-D",properties2,vr,sensitiveVars);
```

Comment:
@derheld42 what I understand from this piece of code is that the single-liner format would be:

```
property1=something + property2=something2 + property3=something3
```

Correct? What would happen if you actually want the `+` symbol to be part of the property? Also, this does not allow combination of single and multi line configuration. Example:

```
property1=something + property2=something2 + property3=something3
property4=something4
```

In any case, this would have to be documented in the inline help of the field. Seems a bit fragile at a first sight.